AGENTS.md

```diff
@@ -0,0 +1,154 @@
+# AI Agent Guide for Podman Development
+
+![PODMAN logo](https://raw.githubusercontent.com/containers/common/main/logos/podman-logo-full-vert.png)
+
+Quick reference guide for AI agents assisting with Podman development.
+
+## Project Overview
+
+**Podman** is a daemonless container engine with Docker-compatible CLI, rootless support, and native pod management.
+
+**Key Technologies**: [Cobra](https://github.com/spf13/cobra), [containers/image](https://github.com/containers/image), [containers/storage](https://github.com/containers/storage), [crun](https://github.com/containers/crun), [Go](https://golang.org), [Netavark](https://github.com/containers/netavark), [runc](https://github.com/opencontainers/runc)
+
+## Quick Start
+
+```bash
+# Build and test
+make binaries           # Build all binaries
+make validatepr         # Format, lint, and validate (required for PRs)
+make localintegration   # Run integration tests
+make localsystem        # Run system tests
+
+# Development tools
+make install.tools      # Install linters and dev tools
+```
+
+## Codebase Structure
+
+```text
+podman/
+├── cmd/podman/               # CLI commands (Cobra framework)
+├── libpod/                   # Core container/pod management (Linux only)
+├── pkg/
+│   ├── api/                  # REST API server
+│   ├── bindings/             # HTTP client (stable API)
+│   ├── domain/               # Business logic layer
+│   │   ├── entities/         # Interfaces and data structures
+│   │   ├── infra/abi/        # Local implementation
+│   │   └── infra/tunnel/     # Remote implementation
+│   └── specgen/              # Container/pod specifications
+├── test/e2e/                 # Integration tests (Ginkgo)
+├── test/system/              # System tests (BATS)
+├── docs/source/markdown/     # Man pages
+└── vendor/                   # Vendored dependencies (DO NOT EDIT)
+```
+
+## Development Patterns
+
+### CLI Command Pattern
+
+```go
+// cmd/podman/command.go
+var commandCmd = &cobra.Command{
+    Use:   "command [options] args",
+    RunE:  commandRun,
+}
+
+func commandRun(cmd *cobra.Command, args []string) error {
+    return registry.ContainerEngine().Command(registry.GetContext(), options)
+}
+```
+
+### Domain Layer Pattern
+
+```go
+// pkg/domain/infra/abi/command.go (local)
+func (ic *ContainerEngine) Command(ctx context.Context, options entities.CommandOptions) error {
+    return ic.Libpod.Command(options)  // Direct libpod call
+}
+
+// pkg/domain/infra/tunnel/command.go (remote)
+func (ic *ContainerEngine) Command(ctx context.Context, options entities.CommandOptions) error {
+    return bindings.Command(ic.ClientCtx, options)  // HTTP API call
+}
+```
+
+## Testing
```

Comment:
No details on what integration vs system tests are, what they are used for, when to use one versus another?